src/sentry/integrations/slack/client.py

```diff
@@ -23,6 +26,26 @@ class SlackClient(IntegrationProxyClient):
     base_url = "https://slack.com/api"
     metrics_prefix = "integrations.slack"
 
+    def __init__(
+        self,
+        integration_id: int | None = None,
+        org_integration_id: int | None = None,
+        verify_ssl: bool = True,
+        logging_context: Mapping[str, Any] | None = None,
+    ) -> None:
+        # The IntegrationProxyClient requires org_integration context to proxy requests properly
+        # but the SlackClient is not often invoked within the context of an organization. This work
+        # around ensures one is always provided
+        self.integration_id = integration_id
+        if not org_integration_id and integration_id is not None:
+            org_integrations = integration_service.get_organization_integrations(
+                integration_id=self.integration_id
+            )
+            if len(org_integrations) > 0:
+                org_integration_id = org_integrations[0].id
```

Comment:
Couldn't this result in messages for one organization going to another organization if they happen to share a slack workspace?  How did we resolve this in the past or has it always been luck of the draw?

Reply:
@markstory In a way yes -- we defer to the integration for access to whatever channel and don't associate channels or any other slack resource with the organization. Since sending messages just requires a channel, as long as that channel exists we send the notification. Whether it was authored by org1 or org2 may be logged, but with respect to the SlackClient, it doesn't care -- it just grabs the token from the integration and sends the payload 🤷